osu.Game.Modes.Taiko/Objects/Drawable/Pieces/CirclePiece.cs

```diff
@@ -0,0 +1,188 @@
+                        },
+                        innerCircleContainer = new CircularContainer
+                        {
+                            Name = "Inner Circle",
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            RelativeSizeAxes = Axes.Both,
+                            Masking = true,
+                            Children = new Framework.Graphics.Drawable[]
+                            {
+                                innerBackground = new Box
+                                {
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                    RelativeSizeAxes = Axes.Both,
+                                },
+                                triangles = new Triangles
+                                {
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                    RelativeSizeAxes = Axes.Both,
+                                }
+                            }
+                        },
+                        new CircularContainer
+                        {
+                            Name = "Ring",
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            RelativeSizeAxes = Axes.Both,
+                            BorderThickness = 8,
+                            BorderColour = Color4.White,
+                            Masking = true,
+                            Children = new[]
+                            {
+                                new Box
+                                {
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                    RelativeSizeAxes = Axes.Both,
+                                    Alpha = 0,
+                                    AlwaysPresent = true
+                                }
+                            }
+                        },
+                        iconContainer = new Container
+                        {
+                            Name = "Icon Container",
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                        }
+                    }
+                },
+            };
+
+            Framework.Graphics.Drawable icon = CreateIcon();
+
+            if (icon != null)
+                iconContainer.Add(icon);
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            backingGlowContainer.EdgeEffect = new EdgeEffect
+            {
+                Type = EdgeEffectType.Glow,
+                Colour = AccentColour,
+                Radius = 8
+            };
+
+            if (KiaiMode)

```

Comment:
remove conditional and make `updateKiaiEffect()`

then in `KiaiMode`'s set, change to
```
if (IsLoaded) updateKiaiEffect();
```